iree/compiler/Conversion/LinalgToLLVMGPU/KernelConfig.cpp

```diff
@@ -36,82 +45,104 @@ static LaunchConfig getOpLaunchConfig(linalg::GenericOp op) {
   SmallVector<int64_t, 4> ts;
   ts.resize(numLoops, 1);
   ts.back() = lowerTs;
-  config.setTileSizes(op, ts, 0);  // Workgroup level.
-  config.setTileSizes(op, {}, 1);  // Subgroup level.
+  tileSizes.push_back(ts);  // Workgroup level.
+  tileSizes.push_back({});  // Subgroup level.
```

Comment:
This is fine now to keep as an NFC change, but this can get strange. Ideally we should just have to separate pipelines. One that goes workgroup -> warp and other that goes workgroup -> threads. That better separates the "programming model used" for lowering. One is using warp-level primitives, and other a thread level code.
Any case, this is more of a hope, and is more of a recommendation.

Reply:
I agree, I was just trying to not change the original behavior since this change is already moving a lot of pieces around.